### PR TITLE
Refactor: StateComputeResult carries full execution result

### PR DESCRIPTION
**Pull request by msmouse**
## Description

Avoids cloning.

## How Has This Been Tested?

existing coverage

## Type of Change
- [x] Refactoring

## Which Components or Systems Does This Change Impact?
- [x] Validator Node

### CODE REVIEW

File: consensus/src/block_storage/block_tree.rs [hunk 1]
```diff
@@ -249,7 +249,8 @@ impl BlockTree {
                        existing_block,
                        block_id,
                        block);
-            checked_verify_eq!(existing_block.compute_result(), block.compute_result());
+            // FIXME(aldenhu): confirm that it's okay to remove
+            // checked_verify_eq!(existing_block.compute_result(), block.compute_result());
```
[the remark this answers is not in the document]
Comment (zekun000) [reply consensus/src/block_storage/block_tree.rs:253]: we're not using the execution result anymore

File: consensus/proptest-regressions/block_storage/block_store_test.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Seeds for failure cases proptest has generated in the past. It is
```
Comment (zekun000) [inline consensus/proptest-regressions/block_storage/block_store_test.txt:1]: what failed here?

File: consensus/consensus-types/src/pipelined_block.rs [hunk 3]
```diff
@@ -44,6 +44,7 @@ pub struct PipelinedBlock {
     /// The state_compute_result is calculated for all the pending blocks prior to insertion to
     /// the tree. The execution results are not persisted: they're recalculated again for the
     /// pending blocks upon restart.
+    #[derivative(PartialEq = "ignore")]
     state_compute_result: StateComputeResult,
```
[the remark this answers is not in the document]
Comment (zekun000) [reply consensus/consensus-types/src/pipelined_block.rs:48]: probably?

File: consensus/proptest-regressions/block_storage/block_store_test.txt (hunk 2 above)
Comment (msmouse) [reply consensus/proptest-regressions/block_storage/block_store_test.txt:1]: the mocks (dummy_xx)

**Review by grao1991 — APPROVED** [review]

**Review by zekun000 — APPROVED** [review]

**Pull request merged**